src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -310,23 +310,27 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 		}
 	}
 
-	public input(data: string): void {
+	public input(data: string, isBinary?: boolean): void {
 		if (this._isDisposed || !this._ptyProcess) {
 			return;
 		}
 		for (let i = 0; i <= Math.floor(data.length / WRITE_MAX_CHUNK_SIZE); i++) {
 			this._writeQueue.push(data.substr(i * WRITE_MAX_CHUNK_SIZE, WRITE_MAX_CHUNK_SIZE));
 		}
-		this._startWrite();
+		this._startWrite(isBinary);
```

Comment:
Let's store isBinary inside the queue, like `{ isBinary: boolean, data: string }`